File: src/main/java/carelog/carelog/journal/domain/JournalTemplateRepository.java
package carelog.carelog.journal.domain;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface JournalTemplateRepository extends JpaRepository<JournalTemplate, Long> {

    Optional<JournalTemplate> findByPublicId(UUID publicId);
    List<JournalTemplate> findAllByStatus(JournalTemplateStatus status);
}

File: src/main/kotlin/carelog/carelog/journal/domain/JournalTemplateRepository.kt
package carelog.carelog.journal.domain

import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface JournalTemplateRepository : JpaRepository<JournalTemplate, Long> {
    fun findByPublicId(publicId: UUID): Optional<JournalTemplate>
    fun findAllByStatus(status: JournalTemplateStatus): List<JournalTemplate>
}
